### ISSUE TITLE
Space in generics - similar to `space-in-parens`

### ISSUE THREAD

**Issue by seahindeniz** [eslint-stylistic/eslint-stylistic#244]
### Clear and concise description of the problem

The following code throws error for the space between `(` and `param1` when `space-in-parens` enabled. But there's no rule that catches the space between `<` and `F,` in the generic scope
```ts
function generic2< F, T> ( param1: F, param2: T) {}
```

### Suggested solution

Having a new rule or adding a support for angle/generic brackets to `space-in-parens` rule. I'm not sure about adding a support for parens rule since the name can be confusing. 
```ts
function generic2<F, T> (param1: F, param2: T) {}
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [X] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by antfu** [comment]
We do have https://eslint.style/rules/default/type-generic-spacing, is that not enough?

**Comment by joshuaavalon** [comment]
@antfu `@stylistic/type-generic-spacing` does not work. It only checks for space after `,` and `=`. 
However, we want to enforce the the space after `<` and before `>`.

**Comment by RayGuo-ergou** [comment]
Yes, all of the following would be valid even with `type-generic-spacing` set to `error`:
`<T >`, `< T >`, `< T>`.

If there's a rule can enforce generic to `<T>` would be very helpful.😁

**Comment by Ericlm** [comment]
I would be happy to submit a PR to fix this, if it's agreed that it should be included inside the `type-generic-spacing` rule 🙂

**Comment by Ericlm** [comment]
Maybe it would be a problem with multiline types like:
```ts
type Foo<
        T = true,
        K = false
      > = T // Deleting spaces before > but keeping newline ?
```
Especially with the `indent` rule that could conflict in this case 🤔

**Comment by Ericlm** [comment]
It could be something like "if < and other char after in the same line", then remove spacing between < and char"? And the same in the other way.
